lib/patch-by-id.js

```diff
@@ -0,0 +1,187 @@
+// Copyright IBM Corp. 2015,2016. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+var ValidationError = require('./validations').ValidationError;
+var assert = require('assert');
+var dao = require('./dao');
```

Comment:
Cyclic dependencies are better to avoid. Please move the helper functions into a standalone file, e.g. `lib/dao/util.js`.